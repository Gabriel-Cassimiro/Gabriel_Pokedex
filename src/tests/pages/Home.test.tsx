import React from "react"
import { render, screen } from "@testing-library/react"
import { GetServerSideProps } from "next"

import Home, { HomeProps, getServerSideProps } from "../../pages/index"
import { TestProvider } from "../utils"
import { charmander, squirtle } from "../mocks/pokemons.mock"

jest.mock("axios", () => {
	const { charmander, squirtle } = require("../mocks/pokemons.mock")
	//No need to use jest-mock-axios library because there is only one single get.
	return {
		get(_: string) {
			return {
				data: {
					pokemon: [charmander, squirtle]
				}
			}
		}
	}
})

describe("Tests Home page", () => {
	function Element({ pokemons }: HomeProps) {
		return (
			<TestProvider>
				<Home pokemons={pokemons} />
			</TestProvider>
		)
	}

	it("should render correctly when receiving pokemon array length bigger than 0", () => {
		const { container } = render(<Element pokemons={[charmander, squirtle]} />)

		expect(screen.getByText("Charmander")).toBeInTheDocument()
		expect(screen.getByText("Squirtle")).toBeInTheDocument()

		expect(container).toMatchSnapshot()
	})

	it("should render correctly when receiving pokemon array length equals 0", () => {
		const { container } = render(<Element pokemons={[]} />)

		expect(screen.getByText("Pokemon nÃ£o existe")).toBeInTheDocument()

		expect(container).toMatchSnapshot()
	})

	it("should return props pokemons", async () => {
		const result = await getServerSideProps({} as GetServerSideProps)

		expect(result).toEqual({
			props: { pokemons: [charmander, squirtle] }
		})
	})
})
