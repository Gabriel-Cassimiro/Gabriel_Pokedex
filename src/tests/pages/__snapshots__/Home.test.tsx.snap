// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Home page should render correctly when receiving pokemon array length bigger than 0 1`] = `
<div>
  <div
    class="homestyles__HeaderContainer-sc-1nzr84s-1 iOMGdc"
  >
    <h1>
       Pokédex 
    </h1>
    <div
      class="searchContainer"
    >
      <svg
        color="black"
        fill="currentColor"
        height="25"
        stroke="currentColor"
        stroke-width="0"
        style="color: black;"
        viewBox="0 0 16 16"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.7 13.3l-3.81-3.83A5.93 5.93 0 0 0 13 6c0-3.31-2.69-6-6-6S1 2.69 1 6s2.69 6 6 6c1.3 0 2.48-.41 3.47-1.11l3.83 3.81c.19.2.45.3.7.3.25 0 .52-.09.7-.3a.996.996 0 0 0 0-1.41v.01zM7 10.7c-2.59 0-4.7-2.11-4.7-4.7 0-2.59 2.11-4.7 4.7-4.7 2.59 0 4.7 2.11 4.7 4.7 0 2.59-2.11 4.7-4.7 4.7z"
          fill-rule="evenodd"
        />
      </svg>
      <input
        class="searchBox"
        data-testid="search-input"
        placeholder="Search..."
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="homestyles__PokemonContainer-sc-1nzr84s-0 klMSwB"
  >
    <div
      class="PokemonSingle"
    >
      <div
        class="image"
      >
        <div
          style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 150px; height: 150px;"
        >
          <noscript />
          <img
            alt="Charmander"
            data-nimg="true"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </div>
      <div
        class="pokeNum"
      >
        #
        004
      </div>
      <h2>
        Charmander
      </h2>
      <div
        class="types"
      >
        <div
          class="homestyles__PokeType-sc-1nzr84s-2 eYwIIJ"
          color="fire"
        >
          Fire
        </div>
      </div>
    </div>
    <div
      class="PokemonSingle"
    >
      <div
        class="image"
      >
        <div
          style="overflow: hidden; box-sizing: border-box; display: inline-block; position: relative; width: 150px; height: 150px;"
        >
          <noscript />
          <img
            alt="Squirtle"
            data-nimg="true"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
      </div>
      <div
        class="pokeNum"
      >
        #
        007
      </div>
      <h2>
        Squirtle
      </h2>
      <div
        class="types"
      >
        <div
          class="homestyles__PokeType-sc-1nzr84s-2 VjBKF"
          color="water"
        >
          Water
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests Home page should render correctly when receiving pokemon array length equals 0 1`] = `
<div>
  <div
    class="homestyles__HeaderContainer-sc-1nzr84s-1 iOMGdc"
  >
    <h1>
       Pokédex 
    </h1>
    <div
      class="searchContainer"
    >
      <svg
        color="black"
        fill="currentColor"
        height="25"
        stroke="currentColor"
        stroke-width="0"
        style="color: black;"
        viewBox="0 0 16 16"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.7 13.3l-3.81-3.83A5.93 5.93 0 0 0 13 6c0-3.31-2.69-6-6-6S1 2.69 1 6s2.69 6 6 6c1.3 0 2.48-.41 3.47-1.11l3.83 3.81c.19.2.45.3.7.3.25 0 .52-.09.7-.3a.996.996 0 0 0 0-1.41v.01zM7 10.7c-2.59 0-4.7-2.11-4.7-4.7 0-2.59 2.11-4.7 4.7-4.7 2.59 0 4.7 2.11 4.7 4.7 0 2.59-2.11 4.7-4.7 4.7z"
          fill-rule="evenodd"
        />
      </svg>
      <input
        class="searchBox"
        data-testid="search-input"
        placeholder="Search..."
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    class="homestyles__PokemonContainer-sc-1nzr84s-0 klMSwB"
  >
    <h2
      class="notFound"
    >
      Pokemon não existe
    </h2>
  </div>
</div>
`;
